### PR TITLE
Add proxy variable for max/min in postprocessors

### LINKED ISSUE #18936: Allow ElementExtremeValue and NodalExtremeValue to use proxies
## Reason
It is a useful operation to be able to apply a max/min postprocessor to variable `x`, and then evaluate another variable at the same position in space. 

## Design
We can add a `proxy_variable` to `ElementExtremeValue` and `NodalExtremeValue` so that we can perform this operation. Then, the max/min values are taken as the `variable` at the location where the `proxy_variable` reaches the specified max/min.

## Impact
New feature.

### PR DESCRIPTION
**Pull request by aprilnovak**
This PR adds an optional `proxy_variable` to the `NodalExtremeValue` and `ElementExtremeValue` postprocessors that finds the value of the `variable` at the point in the domain at which the `proxy_variable` reaches its max/min value. If not specified, the proxy variable is taken equal to the `variable`, such that behavior is unchanged from the default usage of these postprocessors.

## Impact
No expected impact. Changes are compatible with previous behavior.

Closes #18936

### CODE REVIEW

**Comment by aprilnovak** [comment]
Test failure seems unrelated?

File: framework/src/postprocessors/ElementExtremeValue.C [hunk 1]
```diff
@@ -31,16 +31,21 @@ ElementExtremeValue::validParams()
                              "returns the maximum value. 'min' returns "
                              "the minimum value.");
 
+  params.addCoupledVar("proxy_variable",
```
Comment (loganharbour) [inline framework/src/postprocessors/ElementExtremeValue.C:34]: Should probably add a comment here describing what happens when this isn't provided. same for the nodal object

File: framework/src/postprocessors/ElementExtremeValue.C [hunk 2]
```diff
@@ -50,11 +55,11 @@ ElementExtremeValue::initialize()
   switch (_type)
   {
     case MAX:
-      _value = -std::numeric_limits<Real>::max(); // start w/ the min
+      _proxy_value = -std::numeric_limits<Real>::max(); // start w/ the min
```
Comment (loganharbour) [inline framework/src/postprocessors/ElementExtremeValue.C:58]: Initialize both `_value` and `_proxy_value` here, below, and also in the nodal object

File: framework/include/userobject/UserObject.h [hunk 3]
```diff
@@ -201,3 +204,12 @@ UserObject::gatherProxyValueMax(T1 & value, T2 & proxy)
...
 }
+
+template <typename T1, typename T2>
+void
+UserObject::gatherProxyValueMin(T1 & value, T2 & proxy)
+{
+  unsigned int rank;
+  _communicator.minloc(value, rank);
+  _communicator.broadcast(proxy, rank);
+}
```
Comment (loganharbour) [inline framework/include/userobject/UserObject.h:215]:
```suggestion
template <typename T1, typename T2>
void
UserObject::gatherProxyValueMin(const T1 & value, T2 & proxy)
{
  processor_id_type rank;
  _communicator.minloc(value, rank);
  _communicator.broadcast(proxy, rank);
}
```

I _think_ you can make `value` const, but you should definitely use `processor_id_type` for the rank. I know this is just copying another implementation - if you wouldn't mind making this change for the max method as well and add a docstring for both.

**Review by loganharbour — DISMISSED** [review]

**Review by loganharbour — CHANGES_REQUESTED** [review]
Request changes instead of approve... whoops

File: framework/include/userobject/UserObject.h (hunk 3 above)
Comment (aprilnovak) [reply framework/include/userobject/UserObject.h:215]: I can use `processor_id_type`, but I can't make the value const - I get lots of errors like this:

```
/home/anovak/moose/scripts/../libmesh/installed/include/timpi/parallel_implementation.h: In instantiation of ‘void TIMPI::Communicator::maxloc(T&, unsigned int&) const [with T = const double]’:
/home/anovak/moose/framework/include/userobject/UserObject.h:216:3:   required from ‘void UserObject::gatherProxyValueMax(const T1&, T2&) [with T1 = double; T2 = double]’
/home/anovak/moose/framework/src/postprocessors/ElementExtremeValue.C:98:47:   required from here
/home/anovak/moose/scripts/../libmesh/installed/include/timpi/parallel_implementation.h:2409:22: error: use of deleted function ‘TIMPI::DataPlusInt<const double>::DataPlusInt()’
 2409 |       DataPlusInt<T> data_in;
```

File: framework/src/postprocessors/ElementExtremeValue.C [hunk 4]
```diff
@@ -44,7 +45,6 @@ ElementExtremeValue::validParams()
 ElementExtremeValue::ElementExtremeValue(const InputParameters & parameters)
   : ElementVariablePostprocessor(parameters),
     _type((ExtremeType)(int)parameters.get<MooseEnum>("value_type")),
-    _value(_type == 0 ? -std::numeric_limits<Real>::max() : std::numeric_limits<Real>::max()),
```
Comment (loganharbour) [inline framework/src/postprocessors/ElementExtremeValue.C]: I'm going to test this with valgrind - if anyone uses this value before initialize() for some reason, we may get an uninitialized vailure

**Review by loganharbour — APPROVED** [review]

**Pull request merged**